### PR TITLE
Feature/fragmented data statistic

### PR DESCRIPTION
**Pull request by softwarejimenez**
:white_check_mark: I have updated the documentation accordingly.
:white_check_mark: I have read the CONTRIBUTING document.

### Summary

In this example, we illustrate how to access the fragmented data statistcs.

### CODE REVIEW

File: examples/connext_dds/fragmented_data_statistics/README.md [hunk 1]
```diff
@@ -0,0 +1,60 @@
+# Example Code: Fragmented Data Statistics
+
+Example about getting fragmented data statistics.
```
Comment (samuelraeburn) [inline examples/connext_dds/fragmented_data_statistics/README.md:3]:
```suggestion
Example demonstrating how to obtain statistics relating to fragmented data.
```

File: examples/connext_dds/fragmented_data_statistics/README.md [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+| DataReaderProtocolStatus | dropped_fragment_count | The number of DATA_FRAG messages that have been dropped by a DataReader. |
+| DataReaderProtocolStatus | sent_nack_fragment_count | The number of NACK fragments that have been sent from a DataReader to a DataWriter. |
+| DataReaderProtocolStatus | sent_nack_fragment_bytes | The number of NACK fragment bytes that have been sent from a DataReader to a DataWriter. |
+
+For more information about the counters, please refer to the API Reference HTML
+documentation or the [User Manuals](https://community.rti.com/static/documentation/connext-dds/6.1.0/doc/manuals/connext_dds_professional/users_manual/index.htm#users_manual/Statuses_for_DataWriters.htm).
+
+## Example Description
+
+In this example, we illustrate how to access the fragmented data statistics:
```
Comment (samuelraeburn) [inline examples/connext_dds/fragmented_data_statistics/README.md:25]:
```suggestion
In this example, we show how to access the fragmented data statistics:
```

File: examples/connext_dds/fragmented_data_statistics/README.md [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+
+In order to update the statistics we have written large data samples, which is
```
Comment (samuelraeburn) [inline examples/connext_dds/fragmented_data_statistics/README.md:27]:
```suggestion
In order to update the fragmented data statistics we write large data samples, which are
```

File: examples/connext_dds/fragmented_data_statistics/README.md [hunk 4]
```diff
@@ -0,0 +1,60 @@
...
+The configuration we are using results in each sample being made up of 3
+fragments. Assuming no repairs are sent, this means that the fragment counters
+should be three times higher than the sample counters.
+
+In the subscriber application, we retrieve the DataReader Protocol Status and
+log the relevant values. The expected output is:
+
+```plaintext
+Fragmented Data Statistics:
+     received_fragment_count 330
```
Comment (samuelraeburn) [inline examples/connext_dds/fragmented_data_statistics/README.md:39]: why is there a discrepancy between recieved_fragment_count and pushed_fragment_count?

File: examples/connext_dds/fragmented_data_statistics/c++/CMakeLists.txt [hunk 5]
```diff
@@ -0,0 +1,19 @@
+# (c) 2020 Copyright, Real-Time Innovations, Inc. All rights reserved.
```
Comment (samuelraeburn) [inline examples/connext_dds/fragmented_data_statistics/c++/CMakeLists.txt:1]: 2021

File: examples/connext_dds/fragmented_data_statistics/c++/USER_QOS_PROFILES.xml [hunk 6]
```diff
@@ -0,0 +1,48 @@
+<?xml version="1.0"?>
+<!--
+ (c) 2020 Copyright, Real-Time Innovations, Inc.  All rights reserved.
```
Comment (samuelraeburn) [inline examples/connext_dds/fragmented_data_statistics/c++/USER_QOS_PROFILES.xml:3]: 2021

File: examples/connext_dds/fragmented_data_statistics/c++/application.h [hunk 7]
```diff
@@ -0,0 +1,106 @@
+/*
+* (c) Copyright, Real-Time Innovations, 2020.  All rights reserved.
```
Comment (samuelraeburn) [inline examples/connext_dds/fragmented_data_statistics/c++/application.h:2]: 2021

File: examples/connext_dds/fragmented_data_statistics/c++/application.h [hunk 8]
```diff
@@ -0,0 +1,106 @@
+/*
```
Comment (samuelraeburn) [inline examples/connext_dds/fragmented_data_statistics/c++/application.h:1]: Don't think this file should be committed

File: examples/connext_dds/fragmented_data_statistics/c++/fragment.idl [hunk 9]
```diff
@@ -0,0 +1,16 @@
+/*
+ * (c) 2020 Copyright, Real-Time Innovations, Inc.  All rights reserved.
```
Comment (samuelraeburn) [inline examples/connext_dds/fragmented_data_statistics/c++/fragment.idl:2]: 2021

File: examples/connext_dds/fragmented_data_statistics/c++/fragment_publisher.cxx [hunk 10]
```diff
@@ -0,0 +1,211 @@
+/*
+* (c) Copyright, Real-Time Innovations, 2020.  All rights reserved.
```
Comment (samuelraeburn) [inline examples/connext_dds/fragmented_data_statistics/c++/fragment_publisher.cxx:2]: 2021

**Review by samuelraeburn — CHANGES_REQUESTED** [review]
Reviewed  C++, same feedback to other languages

File: examples/connext_dds/fragmented_data_statistics/README.md (hunk 4 above)
Comment (softwarejimenez) [reply examples/connext_dds/fragmented_data_statistics/README.md:39]: It was just copied at a different moment.
I do not think it is that important.

File: examples/connext_dds/fragmented_data_statistics/c++/application.h (hunk 8 above)
Comment (softwarejimenez) [reply examples/connext_dds/fragmented_data_statistics/c++/application.h:1]: It is needed to allow compile the examples.

File: examples/connext_dds/fragmented_data_statistics/c++11/USER_QOS_PROFILES.xml [hunk 11]
```diff
@@ -0,0 +1,48 @@
...
+        <!-- QoS profile used to configure reliable communication between the DataWriter
+             and DataReader created in the example code.
+
+             A QoS profile groups a set of related QoS.
+        -->
+        <qos_profile name="fragment_Profile" base_name="BuiltinQosLib::Generic.StrictReliable" is_default_qos="true">
+            <!-- QoS used to configure the data writer created in the example code -->
+            <datawriter_qos>
+
+                <!-- QoS for fragmenthronous publishing -->
```
Comment (samuelraeburn) [inline examples/connext_dds/fragmented_data_statistics/c++11/USER_QOS_PROFILES.xml:31]:
```suggestion
                <!-- QoS for asynchronous publishing -->
```

File: examples/connext_dds/fragmented_data_statistics/c++/USER_QOS_PROFILES.xml [hunk 12]
```diff
@@ -0,0 +1,48 @@
...
+        <!-- QoS profile used to configure reliable communication between the DataWriter
+             and DataReader created in the example code.
+
+             A QoS profile groups a set of related QoS.
+        -->
+        <qos_profile name="fragment_Profile" base_name="BuiltinQosLib::Generic.StrictReliable" is_default_qos="true">
+            <!-- QoS used to configure the data writer created in the example code -->
+            <datawriter_qos>
+
+                <!-- QoS for fragmenthronous publishing -->
```
Comment (samuelraeburn) [inline examples/connext_dds/fragmented_data_statistics/c++/USER_QOS_PROFILES.xml:31]:
```suggestion
                <!-- QoS for asynchronous publishing -->
```
(and in all other languages)

**Review by samuelraeburn — APPROVED** [review]

**Pull request merged**